sklearn/feature_extraction/hashing.py

```diff
@@ -123,6 +125,15 @@ def fit(self, X=None, y=None):
         """
         # repeat input validation for grid search (which calls set_params)
         self._validate_params(self.n_features, self.input_type)
+
+        # optional if random_state is left to an integer seed
+        if isinstance(self.random_state, int):
+            self.seed_ = self.random_state
+        else:
+            self.random_state = check_random_state(self.random_state)
```

Comment:
you should not modify attributes defined in the __init__ (`self.random_state` here). Instead:
```
random_state = check_random_state(self.random_state)
self.seed_ = int(random_state.get_state()[1][0])
```